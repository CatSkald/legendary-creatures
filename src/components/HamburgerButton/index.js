import styles from "./index.module.scss";

import React from "react";
import PropTypes from "prop-types";

export const HamburgerButton = (props) => {
  return (
    <button
      onClick={props.handleClick}
      className={`${styles.HamburgerButton} ${
        props.isActive ? styles.active : ""
      }`}
      aria-label="Show menu"
    >
      <span></span>
    </button>
  );
};

HamburgerButton.propTypes = {
  handleClick: PropTypes.func.isRequired,
  isActive: PropTypes.bool.isRequired,
};

export default HamburgerButton;
