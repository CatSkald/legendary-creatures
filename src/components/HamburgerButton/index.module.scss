.active {
  &.hamburgerMenu span {
    background: transparent;

    &:before,
    &:after {
      top: 0;
      left: 0;
    }

    &:before {
      transform: rotate(-45deg);
    }

    &:after {
      transform: rotate(45deg);
    }
  }
}

.hamburgerMenu {
  background: transparent;
  display: inline-block;
  width: 50px;
  height: 20px;
  border: none;
  position: absolute;
  border-radius: var(--border-radius);
  top: var(--space-sm);
  left: var(--space);
  cursor: pointer;

  @media screen and (min-width: 768px /*--screen-medium*/) {
    display: none;
  }

  span {
    background: var(--menu-text);
    border-radius: 10px;
    display: inline-block;
    height: 3px;
    width: 90%;
    transform: translate(-50%, -50%);

    &:before,
    &:after {
      content: "";
      background: var(--menu-text);
      display: inline-block;
      height: 3px;
      width: 100%;
      position: absolute;
      left: 0;
      transition: var(--transition);
    }

    &:before {
      top: -10px;
    }

    &:after {
      bottom: -10px;
    }
  }
}
