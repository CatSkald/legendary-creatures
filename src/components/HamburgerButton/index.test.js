import React from "react";
import { render, fireEvent } from "@testing-library/react";

import HamburgerButton from "../HamburgerButton";

describe("HamburgerButton", () => {
  test.each([true, false])("renders correctly when active is %s", (active) => {
    const { container } = render(
      <HamburgerButton handleClick={jest.fn()} isActive={active} />,
    );
    expect(container).toMatchSnapshot();
  });

  test("click invokes handleClick", () => {
    const mockCallback = jest.fn();
    const { container } = render(
      <HamburgerButton handleClick={mockCallback} isActive={true} />,
    );

    fireEvent.click(container.firstChild);

    expect(mockCallback).toHaveBeenCalledTimes(1);
  });
});
