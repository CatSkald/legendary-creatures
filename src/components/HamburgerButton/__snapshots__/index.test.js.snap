// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HamburgerButton renders correctly when active is false 1`] = `
<div>
  <button
    aria-label="Show menu"
    class="HamburgerButton "
  >
    <span />
  </button>
</div>
`;

exports[`HamburgerButton renders correctly when active is true 1`] = `
<div>
  <button
    aria-label="Show menu"
    class="HamburgerButton active"
  >
    <span />
  </button>
</div>
`;
